staging/src/k8s.io/apiserver/pkg/storage/storagebackend/factory/etcd3.go

```diff
@@ -490,3 +509,78 @@ func startDBSizeMonitorPerEndpoint(client *clientv3.Client, interval time.Durati
 		cancel()
 	}, nil
 }
+
+// EtcdClientInterface for testing purposes on checkConsistentListCacheFlagForDepricatedVersion
+type EtcdClientInterface interface {
+	Endpoints() []string
+	Status(ctx context.Context, endpoint string) (*clientv3.StatusResponse, error)
+}
+
+// checkConsistentListCacheFlagForDepricatedVersion check the ConsistentListFromCache flag and
+// evaluate if the version of etcd is suitable for this feature based on these issues:
+//   - https://github.com/etcd-io/etcd/issues/15220
+//   - https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/2340-Consistent-reads-from-cache/README.md#bug-in-etcd-progress-notification
+func checkConsistentListCacheFlagForDepricatedVersion(c EtcdClientInterface, dialTimeout time.Duration) (err error) {
+	// This part only checks if the feature is enabled or no, get default value of feature with this:
+	// featureDefaultValue := utilfeature.DefaultMutableFeatureGate.GetAll()[genericfeatures.ConsistentListFromCache].Default
+	consistentListFromCacheEnabled := utilfeature.DefaultFeatureGate.Enabled(genericfeatures.ConsistentListFromCache)
```

Comment:
I can get the default value of feature with this `utilfeature.DefaultMutableFeatureGate.GetAll()[genericfeatures.ConsistentListFromCache].Default` but i don't know if its ok or needed to use this here or not